### ISSUE TITLE
[Accessibility] Visible label is not associated programmatically - Button (Amazon Pay)

### ISSUE THREAD

**Issue by amberhinds** [woocommerce/woocommerce-gateway-amazon-pay#332]
## Audit Information
This issue came from an audit WooCommerce had Equalize Digital complete in December 2025. I'm moving the issue out of Linear (WooA11Y-424) to this repo.

- Audit ID: WooExt-521
- Component/Page: Checkout
- Affected Element: Select button
- URL where issue was discovered: https://apay-woo.mystagingwebsite.com/checkout/

## WCAG Compliance

- Success Criterion: 1.3.1 Info and Relationships
- WCAG Version: 2
- Level: A
- Severity: High

## Issue Description

The Select button does not specific that it is for selecting the AmazonPay payment method. If multiple payment methods were present, this may cause confusion as to the functionality of the button.

### Impacted Populations

Blind and low vision people, Screen reader users (blind/low vision and neurodivergent people)

## Recommended Fix

Add screen reader text or an aria-label that adds additional context. For example, "Select: AmazonPay payment method".

## Code Example
```
<div class="wc-block-components-radio-control-accordion-option wc-block-components-radio-control-accordion-option--checked-option-highlighted"><label class="wc-block-components-radio-control__option wc-block-components-radio-control__option-checked" for="radio-control-wc-payment-method-options-amazon_payments_advanced"><input id="radio-control-wc-payment-method-options-amazon_payments_advanced" class="wc-block-components-radio-control__input" type="radio" name="radio-control-wc-payment-method-options" aria-describedby="radio-control-wc-payment-method-options-amazon_payments_advanced__content" aria-disabled="false" value="amazon_payments_advanced" checked=""><div class="wc-block-components-radio-control__option-layout"><div class="wc-block-components-radio-control__label-group"><span id="radio-control-wc-payment-method-options-amazon_payments_advanced__label" class="wc-block-components-radio-control__label"><span class="wc-block-components-payment-method-label">Amazon Pay</span></span></div></div></label><div id="radio-control-wc-payment-method-options-amazon_payments_advanced__content" class="wc-block-components-radio-control-accordion-content"><p><a href="#" class="wc-apa-widget-change amazonpay-merchant-shadow-root-parent-element-for-executing-modal-script" id="amazon_change_payment_method" data-wc_apa_change_bind="changePayment">Select</a>Payment Method</p><div class="payment_method_display"><span class="wc-apa-amazon-logo"></span>Visa ****1111 (Amazon Pay)</div></div></div>
```
## Screenshots

https://drive.google.com/file/d/1N0HT3rJkPsAoooWFT8MREzooPu9hjixO/view?usp=drivesdk
